plugins/catalog-backend/src/next/DefaultLocationStore.ts

```diff
@@ -111,6 +111,39 @@ export class DefaultLocationStore implements LocationStore, EntityProvider {
     });
   }
 
+  async refreshLocation(type: string): Promise<string> {
+    if (!this.connection) {
+      throw new Error('location store is not initialized');
+    }
+
+    const location = await this.db<DbLocationsRow>('locations')
+      .where({ type })
+      .select('id');
+
+    if (!location[0]) {
+      throw new NotFoundError(`Found no location with type ${type}`);
+    }
+
+    const id = location[0].id;
+
+    const isRefreshRowExists = await this.db<DbRefreshStateRow>('refresh_state')
+      .where('entity_id', id)
+      .select();
+
+    let result = `Location found, refresh row not found for type ${type}, please wait a while and try again`;
+    if (isRefreshRowExists.length) {
+      const refreshResult = await this.db<DbRefreshStateRow>('refresh_state')
+        .update({ next_update_at: Date.now().toString() }, [id])
+        .where('entity_id', id);
+
+      if (refreshResult.length) {
+        result = `Refresh row for type ${type} updated`;
+      }
+    }
+
+    return result;
```

Comment:
Really not sure about returning a `string` from this really.

Wondering what return type should be here to be honest, think this is going to require some thought and some input from @backstage/maintainers when some of them are back from holiday.

Feels like we could hold on to the connection until the refresh is complete and return the refreshed entity, but that could also open up problems if the refresh takes a while.

Reply:
re: what we're returning, I don't much care about the type but we would like some messaging of some kind.